Update low level HTML pages to null safety (#3596)
dart-lang/site-www: src/_tutorials/web/low-level-html/connect-dart-html.md

| <a href="{{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-html/Element-class.html" target="_blank" rel="noopener">Element</a>   | A subclass of Node; implements a web page element.        |
| <a href="{{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-html/Document-class.html" target="_blank" rel="noopener">Document</a> | Another subclass of Node; implements the document object. |
{: .table}

The dart:core library, which is automatically imported, 
contains many other useful classes, such as:
<a href="{{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-core/List-class.html" target="_blank" rel="noopener">List,</a>
a parameterized class that can specify the type of its members.
An instance of `Element` keeps its list of child elements
in a `List<Element>`.

### Using the querySelector() function

This app's `main()` function contains a single
line of code that is a little like a run-on sentence
with multiple things happening one after another.
Let's deconstruct it.

`querySelector()` is a top-level function provided by the Dart HTML library

> that gets an `Element?` object from the DOM matching the passed in specifier.

Reviewer: I don't think adding more words to this sentence helps... it's already pretty long, and those details are covered below.

```suggestion
that gets an `Element?` object from the DOM.
```

{% prettify dart tag=pre+code %}
[!querySelector!]('#RipVanWinkle')!.text = 'Wake up, sleepy head!';
{% endprettify %}

> In this example, the argument passed to the function, `'#RipVanWinkle'`,
> is a string literal containing the CSS selector that identifies the object.

Reviewer: I like something closer to the original wording, which is talking in general about the querySelector arg, not about the particular arg in this example.

```suggestion
The argument to `querySelector()` is a string
containing a CSS selector that identifies the object.
```

Here's the full HTML code for this app,
assuming that the Dart code is in a file named `main.dart`:

> {% prettify html tag=pre+code %}

Reviewer: Good catch!

Does  <code>```html</code> not work?

Author: It does, but the resulting HTML highlighting does not seem to handle it well at all:

<img width="952" alt="image" src="https://user-images.githubusercontent.com/18372958/142935889-6f8b7aab-3de3-4247-983d-26acfb9aeeac.png">

For now, I'll stick to prettify for these and investigate this separately (https://github.com/dart-lang/site-www/issues/3685)

Reviewer: Ah right. I was encountering this myself recently.

that control the appearance of page elements.
Let's customize the CSS for the mini app.

1. Click **CSS**, at the upper left of DartPad.
   The view switches from Dart code to the (non-existent) CSS code.

2. Add the following CSS code:

>    {% prettify css tag=pre+code %}

Reviewer: Looks much better with the formatting!

But the only time I can remember choosing to use prettify is when I use highlighting. (<code>```css</code> doesn't work?)

Selectors can also be nested.

CSS selectors are important in Dart programs
because you use them with `querySelector()` and `querySelectorAll()`
to get matching elements from the DOM.
Most often Dart programs use ID selectors with `querySelector()`
and class selectors with `querySelectorAll()`.

Here are some examples of CSS selectors:

| Selector type | Example | Description |
|---|---|
| ID selector | `#RipVanWinkle` | Matches a single, unique element |
| HTML element | `p` | Matches all paragraphs |
| HTML element | `h1` | Matches all level-one headers |

> | Class | `_.classname_` | Matches all items with the class _classname_  |

Reviewer: _.classname_ isn't rendering correctly in the second column. Maybe use a non-code font for everything after the period? The following may or may not work ;)
```suggestion
| Class | <code>.<em>classname</em></code> | Matches all items with the class _classname_  |
```

<img class="scale-img-max" src="/tutorials/web/images/css-rule-explained.png"
     alt="The parts of a CSS rule">

In the mini app, the selector `#RipVanWinkle` is an ID selector,

> as signaled by the hashtag (`#`);

Reviewer: Good catch on _hashtag_!
